app/src/main/java/com/android/unio/ui/association/EditAssociation.kt

```diff
@@ -0,0 +1,221 @@
+package com.android.unio.ui.association
+
+import android.util.Log
+import android.widget.Toast
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import com.android.unio.R
+import com.android.unio.model.association.Association
+import com.android.unio.model.association.AssociationCategory
+import com.android.unio.model.association.AssociationViewModel
+import com.android.unio.ui.navigation.NavigationAction
+import com.android.unio.ui.navigation.Screen
+import kotlinx.coroutines.launch
+
+@Composable
+fun EditAssociationScreen(
+    associationId: String,
+    associationViewModel: AssociationViewModel,
+    navigationAction: NavigationAction,
+) {
+  val context = LocalContext.current
+  val associations by associationViewModel.associations.collectAsState()
+  val association = associations.find { it.uid == associationId }
+  if (association != null) {
+    EditAssociationScaffold(association, navigationAction) { newAssociation ->
+      associationViewModel.saveAssociation(
+          newAssociation,
+          null,
+          onSuccess = {
+            Log.d("EditAssociationScreen", "Association saved successfully.")
+            navigationAction.navigateTo(
+                Screen.withParams(Screen.ASSOCIATION_PROFILE, association.uid),
+                Screen.EDIT_ASSOCIATION)
+          },
+          onFailure = {
+            Log.e("EditAssociationScreen", "Failed to save association.")
+            Toast.makeText(
+                    context, context.getString(R.string.save_failed_message), Toast.LENGTH_SHORT)
+                .show()
+          })
+    }
+  } else {
+    Text(
+        text = context.getString(R.string.association_not_found),
+        color = Color.Red,
+        modifier = Modifier.padding(16.dp).testTag("associationNotFound"))
+  }
+}
+
+@Composable
+fun EditAssociationScaffold(
+    association: Association,
+    navigationAction: NavigationAction,
+    onSave: (Association) -> Unit
+) {
+  val context = LocalContext.current
+  val scope = rememberCoroutineScope()
+  var url by remember { mutableStateOf(TextFieldValue(association.url)) }
+  var name by remember { mutableStateOf(TextFieldValue(association.name)) }
+  var fullName by remember { mutableStateOf(TextFieldValue(association.fullName)) }
+  var description by remember { mutableStateOf(TextFieldValue(association.description)) }
+  var image by remember { mutableStateOf(TextFieldValue(association.image)) }
```

Comment:
You should avoid exposing the image url directly to users because that way anyone can put the URL they want, which will make the application make requests to unknown servers on behalf of users.
To change the image of the association, there should only be an image input that then uploads that image to our servers on Firebase, not a direct image URL input field.

Reply:
Ok I will accord that with image uploading